### PR TITLE
fix(estate): Adding advocates for Estate Members under 18

### PR DESCRIPTION
**Pull request by albinagu**
Screenshot 2023-06-28 at 12 58 04

Screenshot 2023-06-26 at 15 39 48

Screenshot 2023-06-26 at 13 34 44

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1204863611010015

### CODE REVIEW

File: libs/application/template-api-modules/src/lib/modules/templates/estate/estate.service.ts [hunk 1]
```diff
@@ -56,6 +60,29 @@ export class EstateTemplateService extends BaseTemplateApiService {
         400,
       )
     }
+
+    const youngheirs = applicationData.estateMembers.filter(
+      (heir) => kennitala.info(heir.nationalId).age < 18,
+    )
+
```
Comment (kksteini) [inline libs/application/template-api-modules/src/lib/modules/templates/estate/estate.service.ts:67]: Hafa smá comment
```suggestion
// Requirements:
//   Flag if any heir is under 18 years old without an advocate/defender
//   Unless official division of estate is taking place, then the incoming data need not be validated
```

File: libs/application/template-api-modules/src/lib/modules/templates/estate/estate.service.ts [hunk 2]
```diff
@@ -145,6 +173,31 @@ export class EstateTemplateService extends BaseTemplateApiService {
...
+        address: 'Gerviheimili 123, 600 Feneyjar',
+        nationalId: '0101302719',
+        email: 'evropa@gervi.com',
+      }
+
+      const fakeChild = {
+        name: 'Gervimaður Undir 18 án málsvara',
+        relation: 'Barn',
+        // This kennitala is for Gervimaður Ísak Miri ÞÍ Jarrah
+        // This test will break in on the 24th of September 2034
```
Comment (kksteini) [inline libs/application/template-api-modules/src/lib/modules/templates/estate/estate.service.ts:190]:
```suggestion
        // This test will stop serving its purpose on the 24th of September 2034
```

File: libs/application/template-api-modules/src/lib/modules/templates/estate/estate.service.ts [hunk 3]
```diff
@@ -56,6 +60,31 @@ export class EstateTemplateService extends BaseTemplateApiService {
...
+
+    const youngheirs = applicationData.estateMembers.filter(
+      (heir) => kennitala.info(heir.nationalId).age < 18,
+    )
+// Requirements:
+//   Flag if any heir is under 18 years old without an advocate/defender
+//   Unless official division of estate is taking place, then the incoming data need not be validated
+    if (youngheirs.length > 0) {
+      if (youngheirs.some((heir) => !heir.advocate)) {
+        if (
```
Comment (kksteini) [inline libs/application/template-api-modules/src/lib/modules/templates/estate/estate.service.ts:72]: Let's not trip the linter :>


```suggestion
    // Requirements:
    //   Flag if any heir is under 18 years old without an advocate/defender
    //   Unless official division of estate is taking place, then the incoming data need not be validated
    if (youngheirs.length > 0) {
      if (youngheirs.some((heir) => !heir.advocate)) {
        if (
```

**Review by kksteini — APPROVED** [review]
LGTM :+1:

**Review by Thorrik58 — APPROVED** [review]

**Pull request merged**